src/Components/Blazor/WebAssembly.Authentication/src/Services/RemoteAuthenticationService.cs

```diff
@@ -120,8 +127,24 @@ public virtual async ValueTask<AccessTokenResult> GetAccessToken()
         /// <inheritdoc />
         public virtual async ValueTask<AccessTokenResult> GetAccessToken(AccessTokenRequestOptions options)
```

Comment:
I think we have some naming issues here now. If I'm reading this correctly, the usage pattern would look like:

```cs
var accessTokenResult = await service.GetAccessToken(options);
if (accessTokenResult.TryGetAccessToken(out var accessToken))
{
    // ...
}
```

Issues:

1. `GetAccessToken` should have an `Async` suffix on its name, right?
2. It's confusing that both of the methods are basically called "get access token". If you got it the first time, why do you have to get it again?

I wish C# had a richer type system for this. The ideal result type would be `AccessToken | AccessTokenRedirection`. But since that's not an option, I'm tending to prefer the low-ceremony option you originally had, e.g.,

```cs
var result = await service.GetAccessTokenAsync(options);
if (result.Success)
{
    // ... use result.Token
}
else
{
    // ... use result.RedirectionUrl
}
```

If we're feeling fancy, we might even consider defining a deconstructor on `AccessTokenResult` so that in all the docs, we'd demonstrate its consumption like this:

```cs
var (token, redirectionUrl) = await service.GetAccessTokenAsync(options);
if (token != null)
{
    // ... use token
}
else
{
    // ... use redirectionUrl
}
```

However this relies on the idea that there will never be any other status besides "got token" and "needs redirection".

Let's discuss this later.

Reply:
Maybe we could even rely on C#8 nullables to communicate the intent more clearly. If `accessTokenResult.Token` and `accessTokenResult.RedirectionUrl` were both marked as nullable, the compiler will force people to at least think about the possibility that the token isn't provided.